cmd/app/app.go

```diff
@@ -158,6 +161,31 @@ func (o *Options) addFlags(cmd *cobra.Command) {
 	cmd.PersistentFlags().StringVar(&o.Client.Docker.LoginURL,
 		"docker-login-url", "https://hub.docker.com/v2/users/login/",
 		"URL to login into docker using username/password.")
+	cmd.PersistentFlags().StringVar(&o.Client.Selfhosted.Username,
+		"selfhosted-username", "",
+		fmt.Sprintf(
+			"Username is authenticate with a selfhosted registry (%s_%s).",
+			envPrefix, envSelfhostedUsername,
+		))
+	cmd.PersistentFlags().StringVar(&o.Client.Selfhosted.Password,
+		"selfhosted-password", "",
+		fmt.Sprintf(
+			"Password is authenticate with a selfhosted registry (%s_%s).",
+			envPrefix, envSelfhostedPassword,
+		))
+	cmd.PersistentFlags().StringVar(&o.Client.Selfhosted.Bearer,
+		"selfhosted-token", "",
+		fmt.Sprintf(
+			"Token is authenticate with a selfhosted registry. Cannot be used with "+
+				"username/password (%s_%s).",
+			envPrefix, envSelfhostedBearer,
+		))
+	cmd.PersistentFlags().StringVar(&o.Client.Selfhosted.LoginURL,
```

Comment:
Can we replace the docker login URL with this?

That was my original intention with that flag. There is only one URL to login to docker hub

Reply:
I see, and I think I get your intention. My reason was: What if I use both docker hub AND some form of selfhosted registry? Actually our own use-case is that we have a chicken & egg problem with some images. Therefore we both use "external" as "internal" images in 1 cluster. 

I'm happy to give more details or discuss this further.

Reply:
I am keen to enable having multiple self-hosted options, but we can cross that bridge in future. For now this should be fine.